metals/src/main/scala/scala/meta/metals/MetalsLanguageServer.scala

```diff
@@ -0,0 +1,272 @@
+          }
+          scribe.info(
+            s"Started: Metals version ${BuildInfo.metalsVersion} in workspace '$workspace' $clientInfo."
+          )
+
+          serverState = ServerState.Initialized(server)
+          metalsService.underlying = server
+
+          server.initialize()
+      }
+    }
+  }
+  private def setupJna(): Unit = {
+    // This is required to avoid the following error:
+    //   java.lang.NoClassDefFoundError: Could not initialize class com.sun.jna.platform.win32.Kernel32
+    //     at sbt.internal.io.WinMilli$.getHandle(Milli.scala:277)
+    //   There is an incompatible JNA native library installed on this system
+    //     Expected: 5.2.2
+    //     Found:    3.2.1
+    System.setProperty("jna.nosys", "true")
+  }
+
+  private def createServer(
+      workspace: AbsolutePath,
+      initializeParams: InitializeParams,
+  ): MetalsLspService = new MetalsLspService(
+    ec = ec,
+    buffers = buffers,
+    workspace = workspace,
+    client = languageClient,
+    initializeParams = initializeParams,
+    charset = charset,
+    time = time,
+    initialConfig = initialConfig,
+    initialUserConfig = initialUserConfig,
+    progressTicks = progressTicks,
+    bspGlobalDirectories = bspGlobalDirectories,
+    sh = sh,
+    isReliableFileWatcher = isReliableFileWatcher,
+    mtagsResolver = mtagsResolver,
+    onStartCompilation = onStartCompilation,
+    classpathSearchIndexer = classpathSearchIndexer,
+  )
+
+  private val isInitialized = new AtomicBoolean(false)
+  override def initialized(
+      params: InitializedParams
+  ): CompletableFuture[Unit] = {
+    // Avoid duplicate `initialized` notifications. During the transition
+    // for https://github.com/natebosch/vim-lsc/issues/113 to get fixed,
+    // we may have users on a fixed vim-lsc version but with -Dmetals.no-initialized=true
+    // enabled.
+    if (isInitialized.compareAndSet(false, true)) {
+      serverState match {
+        case ServerState.Initialized(server) =>
+          server.initialized()
+        case _ =>
+          Future.failed(new Exception)
+      }
+    } else {
+      scribe.warn("Ignoring duplicate 'initialized' notification.")
+      Future.unit
+    }
+  }.recover { case NonFatal(e) =>
+    scribe.error("Unexpected error initializing server", e)
+  }.asJava
+
+  override def shutdown(): CompletableFuture[Unit] = serverState match {
+    case ServerState.Initialized(server) =>
+      server.shutdown()
+    case _ =>
+      Future.unit.asJava
+  }
+
+  override def exit(): Unit = serverState match {
+    case ServerState.ShuttingDown(server) =>
+      server.exit()
+    case _ => ()
+  }
+
+  override val getScalaService: ScalaLspService = metalsService
+
+  @deprecated
+  def getOldMetalsLanguageServer: MetalsLspService = serverState match {
+    case ServerState.Initialized(server) => server
+    case _ => throw new IllegalStateException("Server is not initialized")
+  }
```

Comment:
Used only in tests, I didn't want to alter tests more than it was needed to fix failing ones. Additional work can be done in follow up of this PR

Reply:
I'd say create an issue right away for this then to outline what would need to be changed and to remove this after.